### PR TITLE
soversioning

### PR DESCRIPTION
**Pull request by rhl-**
versioning the pmrrr and ElSuiteSparse

### CODE REVIEW

**Comment by rhl-** [comment]
(In reply to Ryan H. Lewis (rhl) from comment #11)
> I believe the latest spec should work.

<
$ mock -r fedora-rawhide-x86_64 --no-clean --shell 'rpm -q --list elemental-openmpi| grep lib64'
Start: shell
/usr/lib64/openmpi/lib/libEl.so
/usr/lib64/openmpi/lib/libEl.so.0
/usr/lib64/openmpi/lib/libEl.so.86-dev
/usr/lib64/openmpi/lib/libElSuiteSparse.so
/usr/lib64/openmpi/lib/libpmrrr.so

$ mock -r fedora-rawhide-x86_64 --no-clean --shell 'rpm -q --list elemental-mpich| grep lib64'
Start: shell
/usr/lib64/mpich/lib/libEl.so
/usr/lib64/mpich/lib/libEl.so.0
/usr/lib64/mpich/lib/libEl.so.86-dev
/usr/lib64/mpich/lib/libElSuiteSparse.so
/usr/lib64/mpich/lib/libpmrrr.so

'/usr/lib64/mpich/lib/libEl.so' and '/usr/lib64/openmpi/lib/libEl.so' must be packed in 'elemental-mpich-devel' and 'elemental-openmpi-devel' respectively.
>

<
%package openmpi-examples
Summary: OpenMPI variant of Elemental
Group: Development/Libraries
** Requires: %{name}-openmpi = %{version}-%{release} **
%description openmpi-examples
Contains the example drivers built against OpenMPI

Required package is not arched, use
Requires: %{name}-openmpi%{?_isa} = %{version}-%{release}
Requires: %{name}-mpich%{?_isa} = %{version}-%{release}
>

<
$ mock -r fedora-rawhide-x86_64 --no-clean --shell 'rpmlint elemental-mpich'
Start: shell
elemental-mpich.x86_64: E: invalid-soname /usr/lib64/mpich/lib/libElSuiteSparse.so libElSuiteSparse.so
elemental-mpich.x86_64: W: undefined-non-weak-symbol /usr/lib64/mpich/lib/libElSuiteSparse.so sqrt
elemental-mpich.x86_64: W: unused-direct-shlib-dependency /usr/lib64/mpich/lib/libElSuiteSparse.so /usr/lib64/mpich/lib/libmpi.so.12
elemental-mpich.x86_64: E: invalid-soname /usr/lib64/mpich/lib/libpmrrr.so libpmrrr.so
elemental-mpich.x86_64: W: unused-direct-shlib-dependency /usr/lib64/mpich/lib/libpmrrr.so /lib64/libopenblas.so.0
elemental-mpich.x86_64: W: unused-direct-shlib-dependency /usr/lib64/mpich/lib/libpmrrr.so /lib64/libgfortran.so.3
elemental-mpich.x86_64: W: unused-direct-shlib-dependency /usr/lib64/mpich/lib/libpmrrr.so /lib64/libpthread.so.0
elemental-mpich.x86_64: W: unused-direct-shlib-dependency /usr/lib64/mpich/lib/libpmrrr.so /lib64/libqd.so.0
elemental-mpich.x86_64: W: shared-lib-calls-exit /usr/lib64/mpich/lib/libpmrrr.so exit@GLIBC_2.2.5
elemental-mpich.x86_64: W: no-documentation
elemental-mpich.x86_64: W: devel-file-in-non-devel-package /usr/lib64/mpich/lib/libEl.so
elemental-mpich.x86_64: E: library-without-ldconfig-postin /usr/lib64/mpich/lib/libEl.so.86-dev
elemental-mpich.x86_64: E: library-without-ldconfig-postun /usr/lib64/mpich/lib/libEl.so.86-dev
1 packages and 0 specfiles checked; 4 errors, 9 warnings.
Finish: shell

rpmlint shows an sqrt 'undefined-non-weak-symbol' warning, the library looks correctly linked with libm.so.

'unused-direct-shlib-dependency /usr/lib64/mpich/lib/libElSuiteSparse.so /usr/lib64/mpich/lib/libmpi.so.12' looks wrong, too.

**Comment by rhl-** [comment]
We should look into fixing the warnings if at all possible.

**Comment by rhl-** [comment]
This PR addresses:

elemental-mpich.x86_64: E: invalid-soname /usr/lib64/mpich/lib/libpmrrr.so libpmrrr.so
elemental-mpich.x86_64: E: invalid-soname /usr/lib64/mpich/lib/libElSuiteSparse.so libElSuiteSparse.so

and

elemental-mpich.x86_64: E: library-without-ldconfig-postin /usr/lib64/mpich/lib/libEl.so.86-dev
elemental-mpich.x86_64: E: library-without-ldconfig-postun /usr/lib64/mpich/lib/libEl.so.86-dev

I'm unsure about the rest of them.

File: external/pmrrr/CMakeLists.txt [hunk 1]
```diff
@@ -82,6 +82,7 @@ else()
   target_link_libraries(pmrrr 
     ${MPI_C_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${MATH_LIBS})
 endif()
+  set_target_properties(pmrrr PROPERTIES VERSION ${EL_VERSION_MINOR} SOVERSION ${EL_VERSION_MAJOR})
```
Comment (poulson) [inline external/pmrrr/CMakeLists.txt]: What is up with the spacing here?

Comment (rhl-) [inline external/pmrrr/CMakeLists.txt]: looks like github. doesnt show up in vim.

**Pull request merged**